### ISSUE TITLE
Auto-detect container memory

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylla#2522]
If `/sys/fs/cgroup/memory/memory.limit_in_bytes` is present, we should take its value as the limit of the amount of memory available. This is set by `docker run --memory` and likely other container engines.

**Comment by penberg** [comment]
No such file for the Docker container created by our ScyllaDB image:

```
[penberg@nero ~]$ docker run --memory 1G --name some-scylla -d scylladb/scylla
[penberg@nero ~]$ docker exec -it some-scylla /bin/sh
sh-4.2# ls -lh /sys/fs/cgroup/memory/memory.limit_in_bytes
ls: cannot access /sys/fs/cgroup/memory/memory.limit_in_bytes: No such file or directory
```

Perhaps `/sys/fs/cgroup/memory` it needs to be mounted somehow?

**Comment by avikivity** [comment]
Fixed by 472afea6cd3f856d7a47ece8138c149973db8962
